roles/openshift-applier/tasks/process-file.yml

```diff
@@ -0,0 +1,39 @@
+---
+
+- name: "Determine location for the 'file' (if applicable)"
+  vars:
+    path: "{{ file }}"
+  import_tasks: check-file-location.yml
+  when:
+  - file|trim != ''
+
+- name: "Set 'file' values (if applicable)"
+  set_fact:
+    file: "{{ file_path }}"
+    file_f_option: "{{ option_f }}"
+  when:
+  - file|trim != ''
+
+- name: "Include any pre-processing role(s) before applying file and/or template"
+  include_tasks: pre-post-step.yml
+  with_items:
+  - "{{ content.pre_steps | default('') }}"
+  loop_control:
+    loop_var: step
+
+- name: "{{ file_action | capitalize }} OpenShift objects based on static files for '{{ entry.object}} : {{ content.name | default(file | basename) }}'"
+  command: >
+    oc {{ file_action }} {{ target_namespace }} -f {{ file }} {{ (file_action == 'delete') | ternary(' --ignore-not-found', '') }}
+  register: command_result
+  failed_when:
+  - command_result.rc != 0
+  - "'AlreadyExists' not in command_result.stderr"
+  when:
+  - file|trim != ''
+
+- name: "Include any post-processing role(s) after applying file and/or template"
```

Comment:
See comment elsewhere re: doing the pre/post processing before the include of this file.